common/flatpak-dir.c

```diff
@@ -363,14 +378,24 @@ flatpak_remote_state_add_sideload_repo (FlatpakRemoteState *self,
   mfile = g_mapped_file_new (flatpak_file_get_path_cached (summary_path), FALSE, NULL);
   if (mfile != NULL && ostree_repo_open (sideload_repo, NULL, NULL))
     {
+      g_autoptr(GError) local_error = NULL;
       g_autoptr(GBytes) summary_bytes = g_mapped_file_get_bytes (mfile);
       FlatpakSideloadState *ss = g_new0 (FlatpakSideloadState, 1);
 
       ss->repo = g_steal_pointer (&sideload_repo);
       ss->summary = g_variant_ref_sink (g_variant_new_from_bytes (OSTREE_SUMMARY_GVARIANT_FORMAT, summary_bytes, TRUE));
-      g_ptr_array_add (self->sideload_repos, ss);
 
-      g_debug ("Using sideloaded repo %s for remote %s", flatpak_file_get_path_cached (dir), self->remote_name);
+      if (!_validate_summary_for_collection_id (ss->summary, self->collection_id, &local_error))
+        {
+          g_warning ("Sideload repo at path %s not valid for remote %s: %s",
+                     flatpak_file_get_path_cached (dir), self->remote_name, local_error->message);
```

Comment:
I think having this being a warning will be a problem, no?
If you have a sideload repo for one remote any update to another remote would trigger this warning.

Reply:
Right, my mistake. Changed it to a `g_debug` and added a comment.